src/Functions/registerFunctionsString.cpp

```diff
@@ -31,6 +31,7 @@ void registerFunctionAppendTrailingCharIfAbsent(FunctionFactory &);
 void registerFunctionStartsWith(FunctionFactory &);
 void registerFunctionEndsWith(FunctionFactory &);
 void registerFunctionTrim(FunctionFactory &);
+void registerFunctionSparkTrim(FunctionFactory &);
```

Comment:
考虑一下能不能放在local_engine的目录下面，或者把函数的实现提交给CH社区

Reply:
我看看咋放到  local_engine 下面，提交给上游估计没戏，对于这种处理 spark 和 ch直接兼容的脏的逻辑

Reply:
I think it maybe better to add an option argument in current clickhouse trim functions instead of creating new functions

Reply:
I think it's better to make a transform during constructing query plan.

Reply:
I completely agree with what you said, but I feel that it is absolutely impossible to merge into the Clickhouse repo, the reason I said above 

The functions in clickhouse can only trim whitespace, and the syntax of 'trim([[LEADING|TRAILING|BOTH] trim_character FROM] input_string)' is parsed into replaceRegexpAll in ExpressionListParsers.cpp, the logic is completely different from spark trim, sparkTrim is not It is easy to abstract with the trim function, because trim comes from a template class of FunctionStringToString, which only accepts one parameter, so I added a sparkTrim function to the backend, and use different functions according to different parameters.

Reply:
I have moved all the changes into the local_engine directory   @liuneng1994   @taiyang-li